facilitator/src/transport/gcs.rs

```diff
@@ -192,7 +226,13 @@ impl StreamingTransferWriter {
             minimum_upload_chunk_size,
             buffer: Vec::with_capacity(minimum_upload_chunk_size * 2),
             object_upload_position: 0,
-            upload_session_uri: upload_session_uri.to_owned(),
+            upload_session_uri: Url::parse(&upload_session_uri).context(format!(
+                "failed to parse upload_session_uri url: {}",
+                &upload_session_uri
+            ))?,
+            agent: AgentBuilder::new()
+                .timeout(std::time::Duration::from_secs(10))
```

Comment:
```suggestion
                .timeout(Duration::from_secs(10))
```
...and `use std::time::Duration` at the top of the file